kedro/io/data_catalog.py

```diff
@@ -161,16 +112,17 @@ def __init__(  # noqa: PLR0913
         self,
         datasets: dict[str, AbstractDataset] | None = None,
         feed_dict: dict[str, Any] | None = None,
-        dataset_patterns: Patterns | None = None,
+        dataset_patterns: Patterns | None = None,  # Kept for interface compatibility
+        default_pattern: Patterns | None = None,  # Kept for interface compatibility
```

Comment:
Changing the order of arguments is a breaking change as well. That's why `default_pattern` was the last argument. 🥲 So we need to keep them the way they were.